ex1/functions.py

```diff
@@ -65,23 +96,76 @@ def f():
 # compare against. The function should return a list with equal length as the
 # input list containing for each number in the original list either True or
 # False if the number was greater than or equal to n.
+def check_list(l, n):
+    '''
+    :param l: list of numbers
+    :param n: comparing number
+    :return: check each number of l if it greater than n
+    '''
+    r = []
+    for e in l:
+        r.append(e >= n)
+    return r
+
 
 # Define a function named check_list_nth that does the same as check_list but
 # uses every nth element of the input list (including the first one). You will
 # need a third input argument.
+def check_list_nth(l, n, i):
+    '''
+    :param l: list of number
+    :param n: comparing number
+    :param i: step size
+    :return: check each number of l with the step size if it greater than n
+    '''
+    r = []
+    for e in l[::i]:
+        r.append(e >= n)
+    return r
+
 
 # Define a function named add_new_list that takes two inputs. A list l and a
 # second variable x to add to the list. Return a new list containing x as the
 # last element without modifying the original list.
+def add_new_list(l, x):
+    '''
+    :param l: list with objects
+    :param x: object to add on the end of the list
+    :return: a copy of the list with param x added on the end
+    '''
+    r = l.copy()
+    r.append(x)
+    return r
 
 # Define a function named remove_nth that takes a list and removes every nth
 # element (including the first one). Use a keyword named nth to set the default
 # value for nth to 2.
+def remove_nth(l, nth = 2):
+    '''
+    :param l: list with objects
+    :param nth: step size
+    :return: a copy of the list without the first and the nth elements
+    '''
+    r = l.copy()
+    del r[::nth]
+    return r
+
 
 # Define a function named search_n that takes a list and a variable x and
 # searches for x in the list. If the variable is found return the index of the
 # variable in the list and the variable. Otherwise use None for both return
 # values
+def search_n(l, x):
+    '''
+    :param l: list with objects
+    :param x: searched object
+    :return: the index of the variable in the list and the variable
+    '''
+    for idx, val in enumerate(l):
+        if val == x:
```

Comment:
The better and faster solution would be to just let python do the searching. 

See https://github.com/TUW-GEO-python-intro/Exercise-1/pull/20/files#diff-7ec7b04e6cd2fbdbc70b7b64bf731ae8R266